drivers/hmis/spec/requests/hmis/submit_hud_assessments_spec.rb

```diff
@@ -95,6 +123,71 @@
       end
     end
   end
+
+  describe 'Saving and then submitting HUD assessments' do
+    [:INTAKE, :UPDATE, :ANNUAL, :EXIT].each do |role|
+      it "#{role}: sets and updates assessment date and entry/exit dates as appropriate" do
+        definition = Hmis::Form::Definition.find_by(role: role)
+        link_id = definition.assessment_date_item.link_id
+        enrollment_date_updated = e1.date_updated
+        enrollment_entry_date = e1.entry_date
+
+        # Create the initial assessment (save as WIP)
+        new_exit_date = '2005-03-02'
+        input = { **test_input, form_definition_id: definition.id, hud_values: { link_id => new_exit_date } }
+        _resp, result = post_graphql(input: { input: input }) { save_assessment_mutation }
+        assessment_id = result.dig('data', 'saveAssessment', 'assessment', 'id')
+        errors = result.dig('data', 'saveAssessment', 'errors')
+        expect(errors).to be_empty
+        assessment = Hmis::Hud::Assessment.find(assessment_id)
+        expect(assessment).to be_present
+        expect(assessment.assessment_date).to eq(Date.parse(new_exit_date))
+        expect(assessment.assessment_detail.assessment_processor).to be_present
+
+        # entry and exit dates should NOT change on save
+        expect(assessment.enrollment.entry_date).to eq(enrollment_entry_date)
+        expect(assessment.enrollment.exit&.exit_date).to be_nil
+        # expect(assessment.enrollment.date_updated.inspect).to be == enrollment_date_updated.inspect
```

Comment:
I can't get this comparison working. The timestamps ARE the same, but for some reason the actual one is always rounded to fewer decimals:

```
       expected: == "Thu, 09 Feb 2023 16:19:51.791831287 EST -05:00"
            got:    "Thu, 09 Feb 2023 16:19:51.791831000 EST -05:00"
```
and
```
       expected: == "Thu, 09 Feb 2023 16:19:52.810008107 EST -05:00"
            got:    "Thu, 09 Feb 2023 16:19:52.810008000 EST -05:00"
```

Reply:
generally we compare these based on `to_date`, but you could `strftime` both sides, something like `strftime('%c')` to throw out the sub-second differences.

Reply:
The timestamps are only milliseconds off from each other, so `%c` does not give enough specificity. Using `to_datetime` results in the same above error where the last 3 digits are zeros 🤔 

I got the tests passing by using `%Y-%m-%d %T.%3N`, just pushed that up.